docs/source/contributors/index.md

```diff
@@ -6,6 +6,8 @@ This page is intended for people interested in building new or modified function
 
 You can develop Jupyter AI on any system that can run a supported Python version up to and including 3.11, including recent Windows, macOS, and Linux versions.
 
+Each Jupyter AI major version works with only one major version of JupyterLab. Jupyter AI 1.x supports JupyterLab 3.x, and Jupyter AI 2.x supports JupyterLab 4.x. The `main` branch supports the newest released major version of JupyterLab, which, as of July 2023, is 4.
```

Comment:
```suggestion
Each Jupyter AI major version works with only one major version of JupyterLab. Jupyter AI 1.x supports JupyterLab 3.x, and Jupyter AI 2.x supports JupyterLab 4.x.
```

We can omit the "as of July 2023, JupyterLab = 4" clause, since we're not going to commit to updating this sentence regularly, and the latest version of JupyterLab is already well-known and easily discoverable.